src/ORM/DB.php

```diff
@@ -430,6 +430,69 @@ public static function inline_parameters($sql, $parameters)
         return $joined;
     }
 
+
+    /**
+     * Replace 1 parameter with the given value
+     * For example, you can use this to replace a parameter with a column name instead of a literal
+     * @param string $sql The parameterised query
+     * @param int $paramIdx The 0-based position of the parameter
+     * @param array $replacement The value to insert into the queyr
+     * @param bool $skipEscaping Set to true to insert the value as-is, with no escaping. Use with caution!
+     *
+     * @return string
+     */
+    public static function replace_parameter($sql, $paramIdx, $replacement, $skipEscaping = false)
+    {
+        $segments = preg_split('/\?/', $sql);
+        $joined = '';
+        $inString = false;
+        $numSegments = count($segments);
+        $currentParamIdx = 0;
+
+        for ($i = 0; $i < $numSegments; $i++) {
+            $input = $segments[$i];
+            // Append next segment
+            $joined .= $segments[$i];
+            // Don't add placeholder after last segment
+            if ($i === $numSegments - 1) {
```

Comment:
You could adjust the loop condition to `$i < $numSegments - 1`

Reply:
Won't we miss the last segment if we do that?

Reply:
Yep, ignore me 😂

Reply:
That's what I do most of the time any way :wink: